modules/integration/tests-common/integration-test-utils/src/main/java/org/wso2/am/integration/test/impl/ApiProductTestHelper.java

```diff
@@ -218,12 +218,17 @@ private void verifyOperations(List<APIOperationsDTO> actual, List<APIOperationsD
 
         boolean isAPIProductInListing = false;
         int productCount = 0;
-        for (APIInfoDTO apiInfo : apiInfos) {
-            if (apiInfo.getId().equals(apiProductDTO.getId())) {
-                isAPIProductInListing = true;
-                ++productCount;
-                verifyApiInfoDtoWithApiProduct(apiInfo, apiProductDTO);
+        int retries = 0;
+        while(productCount==0 && retries<5){
+            Thread.sleep(5000);
```

Comment:
Shall we have this after the if condition?

Reply:
Done